engine/src/execution_graph/logic_controllers/BatchProcessing.cpp

```diff
@@ -346,15 +346,29 @@ kstatus Projection::run() {
     CodeTimer timer;
 
     std::unique_ptr <ral::cache::CacheData> cache_data = this->input_cache()->pullCacheData();
-    while(cache_data != nullptr ){
-        std::vector<std::unique_ptr <ral::cache::CacheData> > inputs;
-        inputs.push_back(std::move(cache_data));
+    std::vector<std::string> data_names = cache_data->names();
```

Comment:
You need to null check cache_data first

Reply:
also rename `data_names` to `column_names`

Reply:
hmmm, the more i think about it, it should never be null. But to be safe, add something like `RAL_EXPECTS(cache_data != nullptr, "ERROR: Projection::run() first input CacheData was nullptr")`